node_modules/oae-principals/lib/init.js

```diff
@@ -56,8 +56,14 @@ module.exports = function(config, callback) {
 var _ensureSchema = function(callback) {
     // Both user and group information will be stored inside of the Principals CF
     Cassandra.createColumnFamilies({
-        'Principals': 'CREATE TABLE "Principals" ("principalId" text PRIMARY KEY, "tenantAlias" text, "displayName" text, "description" text, "email" text, "emailPreference" text, "visibility" text, "joinable" text, "lastModified" text, "locale" text, "publicAlias" text, "largePictureUri" text, "mediumPictureUri" text, "smallPictureUri" text, "admin:global" text, "admin:tenant" text, "notificationsUnread" text, "notificationsLastRead" text, "acceptedTC" text, "createdBy" text, "created" timestamp, "deleted" timestamp)',
-        'PrincipalsByEmail': 'CREATE TABLE "PrincipalsByEmail" ("email" text PRIMARY KEY, "principalId" text)'
+        'Principals': 'CREATE TABLE "Principals" ("principalId" text PRIMARY KEY, "tenantAlias" text, "displayName" text, "description" text, "email" text, "emailPreference" text, "visibility" text, "joinable" text, "lastModified" text, "locale" text, "publicAlias" text, "largePictureUri" text, "mediumPictureUri" text, "smallPictureUri" text, "admin:global" text, "admin:tenant" text, "notificationsUnread" text, "notificationsLastRead" text, "acceptedTC" text, "createdBy" text, "created" timestamp, "deleted" timestamp, "emailVerified" boolean, "emailChange" text)',
+
+        // Map an email address to user ids. An e-mail address can be used by *multiple* users
+        'PrincipalsByEmail': 'CREATE TABLE "PrincipalsByEmail" ("email" text, "principalId" text, PRIMARY KEY ("email", "principalId")) WITH COMPACT STORAGE',
+
+        // Map email tokens to user ids. We use the email token as the primary key, so a user who
+        // accidentally generated multiple tokens can use each one of them
+        'EmailTokens': 'CREATE TABLE "EmailTokens" ("token" text PRIMARY KEY, "userId" text)'
```

Comment:
`EmailTokens` should be prefixed with `Principals`. I think `PrincipalsEmailToken` sits best with me since a user can only have one pending token.